src/Ql/PdoConnection.php

```diff
@@ -1,130 +1,69 @@
 <?php
 namespace Packaged\Dal\Ql;
 
-use Packaged\Config\ConfigurableInterface;
-use Packaged\Config\ConfigurableTrait;
-use Packaged\Dal\DalResolver;
 use Packaged\Dal\Exceptions\Connection\ConnectionException;
 use Packaged\Dal\Exceptions\Connection\PdoException;
-use Packaged\Dal\IResolverAware;
-use Packaged\Dal\Traits\ResolverAwareTrait;
 use Packaged\Helpers\Strings;
 use Packaged\Helpers\ValueAs;
 
-class PdoConnection
-  implements IQLDataConnection, ConfigurableInterface, ILastInsertId,
-             IResolverAware
+/**
+ * Class PdoConnection
+ *
+ * @property \PDO $_connection
+ */
+class PdoConnection extends AbstractQlConnection
 {
-  use ConfigurableTrait;
-  use ResolverAwareTrait;
-
-  /**
-   * @var \PDO
-   */
-  protected $_connection;
-  protected $_prepareDelayCount = [];
-  protected $_prepareCache = [];
-  protected $_lastConnectTime = 0;
   protected $_emulatedPrepares = false;
+  protected $_prepareDelayCount = [];
   protected $_delayedPreparesCount = null;
-  protected $_inTransaction = false;
-  protected $_lastRetryCount = 0;
-  protected $_maxPreparedStatements = null;
 
-  /**
-   * Open the connection
-   *
-   * @return static
-   *
-   * @throws ConnectionException
-   */
-  public function connect()
+  public function _makeConnection()
   {
-    if(!$this->isConnected())
+    $dsn = $this->_config()->getItem('dsn', null);
+    if($dsn === null)
     {
-      $this->_clearStmtCache();
-
-      $dsn = $this->_config()->getItem('dsn', null);
-      if($dsn === null)
-      {
-        $dsn = sprintf(
-          "mysql:host=%s;dbname=%s;port=%d",
-          $this->_config()->getItem('hostname', '127.0.0.1'),
-          $this->_config()->getItem('database'),
-          $this->_config()->getItem('port', 3306)
-        );
-      }
-
-      $remainingAttempts = ((int)$this->_config()->getItem(
-        'connect_retries',
-        3
-      ));
-
-      while(($remainingAttempts > 0) && ($this->_connection === null))
-      {
-        try
-        {
-          $options = array_replace(
-            $this->_defaultOptions(),
-            ValueAs::arr($this->_config()->getItem('options'))
-          );
+      $dsn = sprintf(
+        "mysql:host=%s;dbname=%s;port=%d",
+        $this->_config()->getItem('hostname', '127.0.0.1'),
+        $this->_selectedDb ?: $this->_config()->getItem('database', ''),
```

Comment:
Does this work if the database name is empty? (i.e. can you put `mysql:host=127.0.0.1;dbname=;port=3306` in the DSN?)